hack/test_e2e.sh

```diff
@@ -40,6 +41,44 @@ if_error_exit() {
     fi
 }
 
+function pass_message {
+    ###########################################################################
+    # Description:                                                            #
+    # show [PASSED] in green and a message as the validation passed.          #
+    #                                                                         #
+    # Arguments:                                                              #
+    #   $1 - message to output                                                #
+    ###########################################################################
+    if ! [ -n "$1" ]; then
+        echo "pass_message() requires a message"
+        exit 1
+    fi
+    GREEN="\e[32m"
+    ENDCOLOR="\e[0m"
+    echo -e "[${GREEN}PASSED${ENDCOLOR}] ${1}"
+}
+
+function detect_container_engine {
+    ###########################################################################
+    # Description:                                                            #
+    # Detect Container Engine, by default it is docker but developers might   #
+    # use real alternatives like podman. The project welcome both.            #
+    #                                                                         #
+    # Arguments:                                                              #
+    #   None                                                                  #
+    ###########################################################################
+
+    echo "* Detecting container engine..."
+    # If docker is not available, let's check if podman exists
+    ${CONTAINER_ENGINE} &> /dev/null
+    if [ "$?" != "0" ]; then
+        CONTAINER_ENGINE="podman"
+        ${CONTAINER_ENGINE} --help &> /dev/null
+        if_error_exit "Unable to detect container engine, try docker or podman"
```

Comment:
How about phrasing the message as 
`The e2e tests currently support docker and podman as the container engine. Please install either of them`?

Reply:
make sense

Reply:
sure, make sense. thanks for the feedback.